src/preact/component/iframe.js

```diff
@@ -0,0 +1,165 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {ContainWrapper, useValueRef} from '../../../src/preact/component';
+import {Loading} from '../../../src/core/loading-instructions';
+import {ReadyState} from '../../../src/ready-state';
+import {forwardRef} from '../../../src/preact/compat';
+import {useAmpContext, useLoading} from '../../../src/preact/context';
+import {
+  useCallback,
+  useEffect,
+  useImperativeHandle,
+  useLayoutEffect,
+  useRef,
+} from '../../../src/preact';
+
+const DEFAULT_MATCHES_MESSAGING_ORIGIN = () => true;
+
+/**
+ * @param {!IframeEmbedDef.Props} props
+ * @param {{current: (!IframeEmbedDef.Api|null)}} ref
+ * @return {PreactDef.Renderable}
+ */
+export function IframeEmbedWithRef(
+  {
+    allow,
+    allowFullScreen,
+    allowTransparency,
+    iframeStyle,
+    name,
+    title,
+    matchesMessagingOrigin = DEFAULT_MATCHES_MESSAGING_ORIGIN,
+    messageHandler,
+    ready = true,
+    loading: loadingProp,
+    onReadyState,
+    sandbox,
+    src,
+    ...rest
+  },
+  ref
+) {
+  const {playable} = useAmpContext();
+  const loading = useLoading(loadingProp);
+  const mount = loading !== Loading.UNLOAD;
+
+  const loadedRef = useRef(false);
+  // The `onReadyStateRef` is passed via a ref to avoid the changed values
+  // of `onReadyState` re-triggering the side effects.
+  const onReadyStateRef = useValueRef(onReadyState);
+  const setLoaded = useCallback(
+    (value) => {
+      if (value !== loadedRef.current) {
+        loadedRef.current = value;
+        const onReadyState = onReadyStateRef.current;
+        onReadyState?.(value ? ReadyState.COMPLETE : ReadyState.LOADING);
+      }
+    },
+    [onReadyStateRef]
+  );
+
+  const iframeRef = useRef(null);
+
+  // Component API: IframeEmbedDef.Api.
+  useImperativeHandle(
+    ref,
+    () => ({
+      // Standard Bento
+      get readyState() {
+        return loadedRef.current ? ReadyState.COMPLETE : ReadyState.LOADING;
+      },
+    }),
+    []
+  );
+
+  // Reset readyState to "loading" when an iframe is unloaded. Has to be
+  // a `useLayoutEffect` to avoid race condition with a future "load" event.
+  // A race condition can happen when a `useEffect` would be executed
+  // after a future "load" is dispatched.
+  useLayoutEffect(() => {
+    if (!mount) {
+      setLoaded(false);
+    }
+  }, [mount, setLoaded]);
+
+  // Pause if the post goes into a "paused" context.
+  useEffect(() => {
+    const iframe = iframeRef.current;
+    if (!playable && iframe) {
+      // Resetting the `src` will reset the iframe and pause it. It will force
```

Comment:
This is a well-known hack and it works great for Instagram. However, heads up: in this simple form it won't be able to cover all use cases. The full support looks like this:

1. If `src && src != 'about:blank' && !src.includes('#') --> do src reset`
2. Otherwise do parent reset.

Src reset is simple:
```
iframe.src = iframe.src
```

Parent reset is also simple:
```
const parent = iframe.parentNode;
parent.removeChild(iframe);
parent.appendChild(iframe);
```

Both do the same thing. Src-reset is faster, but unfortunately it doesn't work in cases that I mention in the conditions above.